databuilder/query_language.py

```diff
@@ -762,8 +762,8 @@ def decorator(cls):
 
         if path.suffix == ".csv":
             rows = csv.read_dataset_csv(path, column_specs)
-        elif path.suffix == ".csv.gz":
-            rows = csv.read_dataset_csv_gz(path, schema)
+        elif path.suffix == ".gz" and path.stem.endswith("csv"):
```

Comment:
Would I be right in thinking that `table_from_file` didn't work with `.csv.gz` files before this change?

```suggestion
        elif p.suffixes == ['.csv', '.gz']
```

Reply:
Correct. The initial failing test was written as more of a sketch but probably shouldn't have been committed as just caused confusion